### PR TITLE
WIP: Dmeerx urological supplies order

### PR DESCRIPTION
**Pull request by yuliumitre**
Create this PR for easy trouble shooting.

### CODE REVIEW

File: UrologicalSupplies/R4/files/UrologicalSuppliesPrepopulation-0.1.0.cql [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+codesystem "SNOMED-CT": 'http://snomed.info/sct'
+codesystem "HCPCS": 'https://bluebutton.cms.gov/resources/codesystem/hcpcs'
+
+valueset "Urinary Incontinence or Chronic Urinary Retention Disorder": 'http://cts.nlm.nih.gov/fhir/ValueSet/2.16.840.1.113762.1.4.1219.70'
+valueset "Urinary Collection and Irrigation System Device or Supply": 'http://cts.nlm.nih.gov/fhir/ValueSet/2.16.840.1.113762.1.4.1219.49'
+valueset "Administrative Gender (HL7 V3)": 'http://cts.nlm.nih.gov/fhir/ValueSet/2.16.840.1.113883.1.11.1'
+valueset "Urinary Collection and Irrigation System Device or Supply	": 'http://cts.nlm.nih.gov/fhir/ValueSet/2.16.840.1.113762.1.4.1219.61'
+valueset "Intermittent Urinary Catheter or Urological Supply for Intermittent Urinary Catheter Supply": 'http://cts.nlm.nih.gov/fhir/ValueSet/2.16.840.1.113762.1.4.1219.59'
+valueset "External Catheter or Urinary Collection Device": 'http://cts.nlm.nih.gov/fhir/ValueSet/2.16.840.1.113762.1.4.1219.62'
+valueset "Miscellaneous Urinary Catheter Supply": 'http://cts.nlm.nih.gov/fhir/ValueSet/2.16.840.1.113762.1.4.1219.63'
```
Comment (yuliumitre) [inline UrologicalSupplies/R4/files/UrologicalSuppliesPrepopulation-0.1.0.cql:17]: I don't see you define those valueset in the Library-R4-Urologicoxxx.json. You need to put those valuesets as "dataRequirement".

File: UrologicalSupplies/R4/files/UrologicalSuppliesPrepopulation-0.1.0.cql [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+
+parameter service_request ServiceRequest
+
+context Patient
+
+////////////////////////////
+context Patient
```
Comment (yuliumitre) [inline UrologicalSupplies/R4/files/UrologicalSuppliesPrepopulation-0.1.0.cql:25]: Only need one context.

File: UrologicalSupplies/R4/files/UrologicalSuppliesRule-0.1.0.cql [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  'Prior Authorization Required.'
+
+define RESULT_Details:
+  'Prior Authorization Required, follow the attached link for information.' 
+
+define RESULT_InfoLink:
+    'https://www.cms.gov/Outreach-and-Education/Medicare-Learning-Network-MLN/MLNProducts/Downloads/ProviderComplianceTipsforVentilators-MLN9725344Print-Friendly.pdf'
+
+define RESULT_QuestionnaireOrderUri:
+    'Questionnaire/UrologicalSuppliesOrder'
```
Comment (yuliumitre) [inline UrologicalSupplies/R4/files/UrologicalSuppliesRule-0.1.0.cql:27]: To follow the naming convention, the order URI should be the full topic name: Questionnaire/UrologicalSupplies

File: UrologicalSupplies/R4/resources/Questionnaire-R4-UrologicalSuppliesOrder.json [hunk 4]
```diff
@@ -0,0 +1,651 @@
...
+    },
+    {
+      "type": "group",
+      "required": false,
+      "text": "Diagnosis",
+      "item": [
+        {
+          "type": "open-choice",
+          "code": [],
+          "extension": [
```
Comment (yuliumitre) [inline UrologicalSupplies/R4/resources/Questionnaire-R4-UrologicalSuppliesOrder.json:151]: This extension should be defined in your xxx-prepopulation.cql. Then you don't need the answerOption[] section.

File: UrologicalSupplies/R4/resources/Questionnaire-R4-UrologicalSuppliesOrder.json [hunk 5]
```diff
@@ -0,0 +1,651 @@
...
+                  }
+                ],
+                "text": "Drop down"
+              }
+            }
+          ],
+          "required": false,
+          "linkId": "5.4",
+          "text": "Order: Miscellaneous urological supplies",
+          "answerOption": [
```
Comment (yuliumitre) [inline UrologicalSupplies/R4/resources/Questionnaire-R4-UrologicalSuppliesOrder.json:527]: Do we need to allow the provider to pick the order here? The device request should contain the order and you probably just need to populate it here.

**Review by lpdecelles — CHANGES_REQUESTED** [review]
Hello Yolanda, it looks like this PR has conflicts.

**Comment by yuliumitre** [comment]
@lpdecelles I created this PR just to help Priyank for his PR with WIP (Work In Progress in the title). I am closing it now.

**Pull request closed without merging**